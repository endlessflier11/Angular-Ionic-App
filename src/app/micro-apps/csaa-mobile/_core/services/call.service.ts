import { Injectable } from '@angular/core';
import { CsaaCommonModule } from '../../csaa-core/csaa-common.module';

@Injectable({
  providedIn: CsaaCommonModule,
})
export class CallService {
  call(phoneNumber: string) {
    if (phoneNumber) {
      phoneNumber = phoneNumber.replace(/ ?x/, ';ext=');
      // This one is working on both IOS and Android in MWG integrated build
      window.location.assign('tel:' + encodeURI(phoneNumber));
    }
  }
}
