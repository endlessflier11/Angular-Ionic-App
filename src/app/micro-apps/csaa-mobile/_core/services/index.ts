export * from './router/router.service';
export * from './config/config.service';
export * from './auth/auth.service';
export * from './sso/sso.service';
export * from './storage/storage.service';
export * from './http/http.service';
export * from './analytics.service';
export * from './policy.service';
export * from './payment.service';
export * from './global-state.service';
export * from './call.service';
export * from './metadata.service';
export * from './pdf-display.service';
export * from './webview/webview.service';
