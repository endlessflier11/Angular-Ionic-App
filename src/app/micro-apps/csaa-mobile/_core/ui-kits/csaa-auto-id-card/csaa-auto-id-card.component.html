<div class="poi-card" *ngIf="!!customerSearch" (click)="toggle.emit(vehicle)">
  <ion-grid>
    <ion-row class="row--heading">
      <ion-col size="auto" class="ion-text-start card-logo">
        <img src="/assets/csaa-mobile/vectors/aaa-logo.svg" />
      </ion-col>
      <ion-col size="10">
        <p>{{ cardContent.headingLine1 }}</p>
        <p>{{ cardContent.headingLine2 }}</p>
        <p>{{ cardContent.headingLine3 }}</p>
      </ion-col>
    </ion-row>
    <ion-row class="row--title">
      <ion-col>
        <h1>{{ cardContent.title }}</h1>
        <ng-container *ngIf="cardContent.showSubtitle">
          <span class="subtitle">PROOF OF INSURANCE</span>
        </ng-container>
      </ion-col>
    </ion-row>
    <ion-row class="row--content">
      <ion-col size="8">
        <p>
          <span>NAMED INSURED</span>
          <span *ngFor="let name of cardContent.namedInsureds" data-test="named-insured">
            {{ name }}
          </span>
        </p>
      </ion-col>
      <ion-col size="4">
        <p>
          <span>POLICY #</span>
          <span data-test="policy-card-number">
            {{ cardContent.policyNumber }}
          </span>
        </p>
      </ion-col>
    </ion-row>
    <ion-row class="row--content">
      <ion-col size="8">
        <p>
          <span>VEHICLE YEAR / MAKE / MODEL</span>
          <span data-test="vehicle-title">
            {{ cardContent.vehicleTitle }}
          </span>
        </p>
      </ion-col>
      <ion-col size="4">
        <p>
          <span>EFFECTIVE DATE</span>
          <span data-test="effective-date">
            {{ cardContent.effectiveDate }}
          </span>
        </p>
      </ion-col>
    </ion-row>
    <ion-row class="row--content">
      <ion-col size="8">
        <p>
          <span>VEHICLE ID#</span>
          <span data-test="vehicle-Id">
            {{ cardContent.vehicleId }}
          </span>
        </p>
      </ion-col>
      <ion-col size="4">
        <p>
          <span>EXPIRATION DATE</span>
          <span data-test="expiration-date">
            {{ cardContent.expirationDate }}
          </span>
        </p>
      </ion-col>
    </ion-row>
    <ion-row *ngIf="cardContent.showCompliance" class="row--content">
      <ion-col size="8">
        <p>
          <span data-test="compliance">
            {{ cardContent.compliance }}
          </span>
        </p>
      </ion-col>
    </ion-row>
    <ion-row *ngIf="cardContent.showExcluded" class="row--content">
      <ion-col size="8">
        <p>
          <span>EXCLUDED DRIVERS</span>
          <span data-test="excluded-drivers">
            {{ cardContent.excludedDrivers }}
          </span>
        </p>
      </ion-col>
    </ion-row>
    <ion-row *ngIf="cardContent.showCustomerAddress" class="row--content">
      <ion-col size="8">
        <p>
          <span>ADDRESS</span>
        </p>
        <ion-row>
          <span data-test="customer-address-1">
            {{ cardContent.addressForCustomer1 }}
          </span>
        </ion-row>
        <ion-row>
          <span data-test="customer-address-2">
            {{ cardContent.addressForCustomer2 }}
          </span>
        </ion-row>
        <ion-row>
          <span data-test="customer-city">
            {{ cardContent.addressForCustomerCity }}, {{ riskState }}
            {{ cardContent.addressForCustomerZip }}
          </span>
        </ion-row>
      </ion-col>
    </ion-row>
  </ion-grid>
</div>
