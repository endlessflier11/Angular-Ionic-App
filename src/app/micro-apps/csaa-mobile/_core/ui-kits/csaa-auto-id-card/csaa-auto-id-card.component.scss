@import '../../../_scss/variables';

:host {
  --ion-grid-column-padding: 5px;
  --ion-grid-column-padding-xs: 5px;
  --ion-grid-column-padding-sm: 5px;
  --ion-grid-column-padding-md: 5px;
  --ion-grid-column-padding-lg: 5px;
  --ion-grid-column-padding-xl: 5px;
}

.poi-card {
  border: none !important;
  padding: 10px 15px !important;
  background: $csaa-auto-id-card-bg-secondary;
  background: linear-gradient(
    90deg,
    $csaa-auto-id-card-bg-secondary,
    $csaa-auto-id-card-bg-primary
  );
  font-size: 9px;
  line-height: 1.6;
  color: white;
  border-radius: 10px;

  ion-grid {
    padding: 0;
  }

  p {
    margin: 0;
    font-size: inherit;
  }

  .row {
    &--heading {
      font-size: inherit;
      flex-wrap: nowrap;

      ion-col {
        padding: 0;
      }

      .card-logo {
        margin-right: 5px;
        padding: 0;
        justify-content: center;
        display: flex;
        flex-direction: column;

        img {
          width: 45px;
        }
      }
    }

    &--title {
      text-align: center;
      white-space: pre-line;

      h1 {
        margin: 5px 0;
        font-size: 14px;
        font-weight: $csaa-active-font-weight;
      }

      .subtitle {
        margin: -5px;
        display: block;
        white-space: nowrap;
        font-weight: $csaa-active-font-weight;
      }
    }

    &--content {
      ion-col {
        font-size: inherit;

        p {
          span {
            &:nth-child(1) {
              display: block;
              white-space: nowrap;
              font-weight: $csaa-active-font-weight;
            }

            &:not(:nth-child(1)) {
              display: block;
              font-weight: $csaa-light-font-weight;
            }
          }
        }
      }

      &:last-child ion-col {
        //padding-bottom: 0;
      }
    }
  }
}
