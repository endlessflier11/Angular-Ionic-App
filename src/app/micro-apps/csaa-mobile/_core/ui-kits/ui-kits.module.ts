import { NgModule } from '@angular/core';
import { CommonModule, DatePipe } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { IonicModule } from '@ionic/angular';
import { ThemeIconComponent } from './theme-icon/theme-icon.component';
import { WorkingHoursComponent } from './working-hours/working-hours.component';
import { MyPolicyCardComponent } from './my-policy-card/my-policy-card.component';
import { SkeletonCardComponent } from './skeleton-card/skeleton-card.component';
import { BackButtonComponent } from './back-button/back-button.component';
import { PhonePipe } from './pipes/phone-number.pipe';
import { CsaaAutoIdCardComponent } from './csaa-auto-id-card/csaa-auto-id-card.component';
import { CsaaLoadingComponent } from './csaa-loading/csaa-loading.component';
import { ContentRendererComponent } from './content-renderer/content-renderer.component';
import { NationalCustomAlertModalComponent } from './national-custom-alert/national-custom-alert-modal';
import { SafeDataPipe } from './pipes/safe-data/safe-data.pipe';
import { DocumentActionModalComponent } from './document-action-modal/document-action-modal.component';
import { CsaaPolicySubtitlesComponent } from './csaa-policy-subtitles/csaa-policy-subtitles.component';
import { PaperlessTermsComponent } from './paperless-terms/paperless-terms.component';
import { PaperlessTermsModalComponent } from './paperless-terms-modal/paperless-terms-modal.component';
import { AppleWalletButtonComponent } from './apple-wallet-button/apple-wallet-button.component';

const UI_COMPONENTS = [
  ThemeIconComponent,
  WorkingHoursComponent,
  MyPolicyCardComponent,
  SkeletonCardComponent,
  BackButtonComponent,
  CsaaAutoIdCardComponent,
  CsaaLoadingComponent,
  ContentRendererComponent,
  NationalCustomAlertModalComponent,
  DocumentActionModalComponent,
  CsaaPolicySubtitlesComponent,
  PaperlessTermsComponent,
  PaperlessTermsModalComponent,
  AppleWalletButtonComponent,
];

const UI_PIPES = [PhonePipe, SafeDataPipe];

@NgModule({
  imports: [CommonModule, ReactiveFormsModule, IonicModule],
  declarations: [...UI_COMPONENTS, ...UI_PIPES],
  exports: [...UI_COMPONENTS, ...UI_PIPES],
  providers: [DatePipe],
})
export class UiKitsModule {}
