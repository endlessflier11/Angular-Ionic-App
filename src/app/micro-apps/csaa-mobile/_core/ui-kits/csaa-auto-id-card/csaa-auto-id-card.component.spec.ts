import { CommonModule, DatePipe } from '@angular/common';
import { HttpTestingController } from '@angular/common/http/testing';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ConfigMockService, deepCopy, generatePolicy, PageTestingModule } from '@app/testing';
import { IonicModule } from '@ionic/angular';
import { EMPLOYEE_SEARCH } from 'src/testing/fixtures/make-payments-page/employee-search.fixture';
import { DriverCoverageType } from '../../interfaces/driver.interface';
import { AnalyticsService, ConfigService } from '../../services';
import { UiKitsModule } from '../ui-kits.module';
import { CsaaAutoIdCardComponent } from './csaa-auto-id-card.component';

describe('CsaaAutoIdCardComponent', () => {
  let component: CsaaAutoIdCardComponent;
  let fixture: ComponentFixture<CsaaAutoIdCardComponent>;
  let httpTestingController: HttpTestingController;
  const realDateNow = Date.now.bind(global.Date);

  beforeEach(async () => {
    TestBed.configureTestingModule({
      declarations: [CsaaAutoIdCardComponent],
      providers: [
        {
          provide: AnalyticsService,
          useValue: {
            trackEvent: jest.fn(),
          },
        },
        { provide: ConfigService, useClass: ConfigMockService },
        {
          provide: DatePipe,
        },
      ],
      imports: [
        CommonModule,
        IonicModule,
        UiKitsModule,
        PageTestingModule.withConfig({ providesStorage: true }),
      ],
    }).compileComponents();

    httpTestingController = TestBed.inject(HttpTestingController);
    fixture = TestBed.createComponent(CsaaAutoIdCardComponent);
    component = fixture.componentInstance;
    component.customerSearch = EMPLOYEE_SEARCH;

    component.policyNumber = 'WYSS910014303';

    component.termEffectiveDate = '2021-05-17';

    component.termExpirationDate = '2021-11-17';

    component.riskState = 'WY';

    const policy = generatePolicy({
      vehicles: [
        {
          id: 'WAUKJAFM8AA061319',
          vin: 'WAUKJAFM8AA061319',
          name: 'Lamborghini Gallardo',
          coverages: [],
          riskFactors: { waivedLiability: true },
        },
      ],
    });

    component.vehicle = policy.vehicles[0];

    component.insureds = [
      {
        id: 'kwgr0tN5YFM8ApZjQotDKg',
        isActive: true,
        firstName: 'JEAN',
        lastName: 'BRANDOS',
        primary: true,
      },
    ];

    component.drivers = [
      {
        id: 'WAUKJAFM8AA061319',
        fullName: 'Judy Quinn',
        dob: realDateNow,
        coverageType: DriverCoverageType.Excluded,
        coverages: [],
      },
    ];
    component.ngOnChanges({});

    fixture.detectChanges();
  });

  afterEach(() => {
    global.Date.now = realDateNow;
    httpTestingController.verify();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should create a snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should match a snapshot for KY', () => {
    const customerSearch = deepCopy(EMPLOYEE_SEARCH);
    customerSearch.policies[0].riskState = 'KY';
    component.customerSearch = customerSearch;
    component.riskState = 'KY';
    component.ngOnChanges({});
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should match a snapshot for NV', () => {
    const customerSearch = deepCopy(EMPLOYEE_SEARCH);
    customerSearch.policies[0].riskState = 'NV';
    component.customerSearch = customerSearch;
    component.riskState = 'NV';
    component.ngOnChanges({});
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should match a snapshot for NJ', () => {
    const customerSearch = deepCopy(EMPLOYEE_SEARCH);
    customerSearch.policies[0].riskState = 'NJ';
    component.customerSearch = customerSearch;
    component.riskState = 'NJ';
    component.ngOnChanges({});
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should match a snapshot for CA', () => {
    const customerSearch = deepCopy(EMPLOYEE_SEARCH);
    customerSearch.policies[0].riskState = 'CA';
    component.customerSearch = customerSearch;
    component.riskState = 'CA';
    component.ngOnChanges({});
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
  
  it('should match a snapshot for OK', () => {
    const customerSearch = deepCopy(EMPLOYEE_SEARCH);
    customerSearch.policies[0].riskState = 'OK';
    component.customerSearch = customerSearch;
    component.riskState = 'OK';
    component.drivers[0].coverageType = DriverCoverageType.Excluded;
    component.ngOnChanges({});
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
