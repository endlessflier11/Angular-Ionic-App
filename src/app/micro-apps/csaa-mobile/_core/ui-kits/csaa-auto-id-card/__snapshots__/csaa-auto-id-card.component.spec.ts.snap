// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsaaAutoIdCardComponent should create a snapshot 1`] = `
<csaa-auto-id-card
  __ngContext__={[Function LRootView]}
  cardContent={[Function Object]}
  customerSearch={[Function Object]}
  datePipe={[Function DatePipe]}
  drivers={[Function Array]}
  insureds={[Function Array]}
  policyNumber={[Function String]}
  riskState={[Function String]}
  termEffectiveDate={[Function String]}
  termExpirationDate={[Function String]}
  toggle={[Function EventEmitter_]}
  vehicle={[Function Object]}
>
  <div
    class="poi-card"
  >
    <ion-grid>
      <ion-row
        class="row--heading"
      >
        <ion-col
          class="ion-text-start card-logo"
          size="auto"
        >
          <img
            src="/assets/csaa-mobile/vectors/aaa-logo.svg"
          />
        </ion-col>
        <ion-col
          size="10"
        >
          <p>
            CSAA Fire & Casualty Insurance Company / NAIC #10921
          </p>
          <p>
            PO Box 24511, Oakland, CA 94623-9865
          </p>
          <p>
            800-207-3618
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--title"
      >
        <ion-col>
          <h1>
            WY Insurance Identification Card
          </h1>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              NAMED INSURED
            </span>
            <span
              data-test="named-insured"
            >
               Jean Brandos 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              POLICY #
            </span>
            <span
              data-test="policy-card-number"
            >
               WYSS910014303 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              VEHICLE YEAR / MAKE / MODEL
            </span>
            <span
              data-test="vehicle-title"
            >
               Lamborghini Gallardo 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              EFFECTIVE DATE
            </span>
            <span
              data-test="effective-date"
            >
               May 17, 2021 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              VEHICLE ID#
            </span>
            <span
              data-test="vehicle-Id"
            >
               WAUKJAFM8AA061319 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              EXPIRATION DATE
            </span>
            <span
              data-test="expiration-date"
            >
               Nov 17, 2021 
            </span>
          </p>
        </ion-col>
      </ion-row>
    </ion-grid>
  </div>
</csaa-auto-id-card>
`;

exports[`CsaaAutoIdCardComponent should match a snapshot for CA 1`] = `
<csaa-auto-id-card
  __ngContext__={[Function LRootView]}
  cardContent={[Function Object]}
  customerSearch={[Function Object]}
  datePipe={[Function DatePipe]}
  drivers={[Function Array]}
  insureds={[Function Array]}
  policyNumber={[Function String]}
  riskState={[Function String]}
  termEffectiveDate={[Function String]}
  termExpirationDate={[Function String]}
  toggle={[Function EventEmitter_]}
  vehicle={[Function Object]}
>
  <div
    class="poi-card"
  >
    <ion-grid>
      <ion-row
        class="row--heading"
      >
        <ion-col
          class="ion-text-start card-logo"
          size="auto"
        >
          <img
            src="/assets/csaa-mobile/vectors/aaa-logo.svg"
          />
        </ion-col>
        <ion-col
          size="10"
        >
          <p>
            CSAA Fire & Casualty Insurance Company / NAIC #10921
          </p>
          <p>
            PO Box 24511, Oakland, CA 94623-9865
          </p>
          <p>
            800-207-3618
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--title"
      >
        <ion-col>
          <h1>
            CA Insurance Identification Card
          </h1>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              NAMED INSURED
            </span>
            <span
              data-test="named-insured"
            >
               Jean Brandos 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              POLICY #
            </span>
            <span
              data-test="policy-card-number"
            >
               WYSS910014303 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              VEHICLE YEAR / MAKE / MODEL
            </span>
            <span
              data-test="vehicle-title"
            >
               Lamborghini Gallardo 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              EFFECTIVE DATE
            </span>
            <span
              data-test="effective-date"
            >
               May 17, 2021 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              VEHICLE ID#
            </span>
            <span
              data-test="vehicle-Id"
            >
               WAUKJAFM8AA061319 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              EXPIRATION DATE
            </span>
            <span
              data-test="expiration-date"
            >
               Nov 17, 2021 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span
              data-test="compliance"
            >
               This insurance complies with CVC ยง16056 or ยง16500.5 
            </span>
          </p>
        </ion-col>
      </ion-row>
    </ion-grid>
  </div>
</csaa-auto-id-card>
`;

exports[`CsaaAutoIdCardComponent should match a snapshot for KY 1`] = `
<csaa-auto-id-card
  __ngContext__={[Function LRootView]}
  cardContent={[Function Object]}
  customerSearch={[Function Object]}
  datePipe={[Function DatePipe]}
  drivers={[Function Array]}
  insureds={[Function Array]}
  policyNumber={[Function String]}
  riskState={[Function String]}
  termEffectiveDate={[Function String]}
  termExpirationDate={[Function String]}
  toggle={[Function EventEmitter_]}
  vehicle={[Function Object]}
>
  <div
    class="poi-card"
  >
    <ion-grid>
      <ion-row
        class="row--heading"
      >
        <ion-col
          class="ion-text-start card-logo"
          size="auto"
        >
          <img
            src="/assets/csaa-mobile/vectors/aaa-logo.svg"
          />
        </ion-col>
        <ion-col
          size="10"
        >
          <p>
            CSAA Fire & Casualty Insurance Company / NAIC #10921
          </p>
          <p>
            PO Box 24511, Oakland, CA 94623-9865
          </p>
          <p>
            800-207-3618
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--title"
      >
        <ion-col>
          <h1>
            COMMONWEALTH OF KENTUCKY
          </h1>
          <span
            class="subtitle"
          >
            PROOF OF INSURANCE
          </span>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              NAMED INSURED
            </span>
            <span
              data-test="named-insured"
            >
               Jean Brandos 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              POLICY #
            </span>
            <span
              data-test="policy-card-number"
            >
               WYSS910014303 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              VEHICLE YEAR / MAKE / MODEL
            </span>
            <span
              data-test="vehicle-title"
            >
               Lamborghini Gallardo 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              EFFECTIVE DATE
            </span>
            <span
              data-test="effective-date"
            >
               May 17, 2021 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              VEHICLE ID#
            </span>
            <span
              data-test="vehicle-Id"
            >
               WAUKJAFM8AA061319 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              EXPIRATION DATE
            </span>
            <span
              data-test="expiration-date"
            >
               Nov 17, 2021 
            </span>
          </p>
        </ion-col>
      </ion-row>
    </ion-grid>
  </div>
</csaa-auto-id-card>
`;

exports[`CsaaAutoIdCardComponent should match a snapshot for NJ 1`] = `
<csaa-auto-id-card
  __ngContext__={[Function LRootView]}
  cardContent={[Function Object]}
  customerSearch={[Function Object]}
  datePipe={[Function DatePipe]}
  drivers={[Function Array]}
  insureds={[Function Array]}
  policyNumber={[Function String]}
  riskState={[Function String]}
  termEffectiveDate={[Function String]}
  termExpirationDate={[Function String]}
  toggle={[Function EventEmitter_]}
  vehicle={[Function Object]}
>
  <div
    class="poi-card"
  >
    <ion-grid>
      <ion-row
        class="row--heading"
      >
        <ion-col
          class="ion-text-start card-logo"
          size="auto"
        >
          <img
            src="/assets/csaa-mobile/vectors/aaa-logo.svg"
          />
        </ion-col>
        <ion-col
          size="10"
        >
          <p>
            CSAA Fire & Casualty Insurance Company / NAIC #10921
          </p>
          <p>
            PO Box 24511, Oakland, CA 94623-9865
          </p>
          <p>
            800-207-3618
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--title"
      >
        <ion-col>
          <h1>
            STATE OF NEW JERSEY 
   INSURANCE IDENTIFICATION CARD
          </h1>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              NAMED INSURED
            </span>
            <span
              data-test="named-insured"
            >
               Jean Brandos 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              POLICY #
            </span>
            <span
              data-test="policy-card-number"
            >
               WYSS910014303 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              VEHICLE YEAR / MAKE / MODEL
            </span>
            <span
              data-test="vehicle-title"
            >
               Lamborghini Gallardo 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              EFFECTIVE DATE
            </span>
            <span
              data-test="effective-date"
            >
               May 17, 2021 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              VEHICLE ID#
            </span>
            <span
              data-test="vehicle-Id"
            >
               WAUKJAFM8AA061319 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              EXPIRATION DATE
            </span>
            <span
              data-test="expiration-date"
            >
               Nov 17, 2021 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              ADDRESS
            </span>
          </p>
          <ion-row>
            <span
              data-test="customer-address-1"
            >
               30 N Piney Rd 
            </span>
          </ion-row>
          <ion-row>
            <span
              data-test="customer-address-2"
            >
            </span>
          </ion-row>
          <ion-row>
            <span
              data-test="customer-city"
            >
               Story, NJ 82832 
            </span>
          </ion-row>
        </ion-col>
      </ion-row>
    </ion-grid>
  </div>
</csaa-auto-id-card>
`;

exports[`CsaaAutoIdCardComponent should match a snapshot for NV 1`] = `
<csaa-auto-id-card
  __ngContext__={[Function LRootView]}
  cardContent={[Function Object]}
  customerSearch={[Function Object]}
  datePipe={[Function DatePipe]}
  drivers={[Function Array]}
  insureds={[Function Array]}
  policyNumber={[Function String]}
  riskState={[Function String]}
  termEffectiveDate={[Function String]}
  termExpirationDate={[Function String]}
  toggle={[Function EventEmitter_]}
  vehicle={[Function Object]}
>
  <div
    class="poi-card"
  >
    <ion-grid>
      <ion-row
        class="row--heading"
      >
        <ion-col
          class="ion-text-start card-logo"
          size="auto"
        >
          <img
            src="/assets/csaa-mobile/vectors/aaa-logo.svg"
          />
        </ion-col>
        <ion-col
          size="10"
        >
          <p>
            CSAA Fire & Casualty Insurance Company / NAIC #10921
          </p>
          <p>
            PO Box 24511, Oakland, CA 94623-9865
          </p>
          <p>
            800-207-3618
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--title"
      >
        <ion-col>
          <h1>
            NV Insurance Identification Card
          </h1>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              NAMED INSURED
            </span>
            <span
              data-test="named-insured"
            >
               Jean Brandos 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              POLICY #
            </span>
            <span
              data-test="policy-card-number"
            >
               WYSS910014303 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              VEHICLE YEAR / MAKE / MODEL
            </span>
            <span
              data-test="vehicle-title"
            >
               Lamborghini Gallardo 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              EFFECTIVE DATE
            </span>
            <span
              data-test="effective-date"
            >
               May 17, 2021 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              VEHICLE ID#
            </span>
            <span
              data-test="vehicle-Id"
            >
               WAUKJAFM8AA061319 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              EXPIRATION DATE
            </span>
            <span
              data-test="expiration-date"
            >
               Nov 17, 2021 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              ADDRESS
            </span>
          </p>
          <ion-row>
            <span
              data-test="customer-address-1"
            >
               30 N Piney Rd 
            </span>
          </ion-row>
          <ion-row>
            <span
              data-test="customer-address-2"
            >
            </span>
          </ion-row>
          <ion-row>
            <span
              data-test="customer-city"
            >
               Story, NV 82832 
            </span>
          </ion-row>
        </ion-col>
      </ion-row>
    </ion-grid>
  </div>
</csaa-auto-id-card>
`;

exports[`CsaaAutoIdCardComponent should match a snapshot for OK 1`] = `
<csaa-auto-id-card
  __ngContext__={[Function LRootView]}
  cardContent={[Function Object]}
  customerSearch={[Function Object]}
  datePipe={[Function DatePipe]}
  drivers={[Function Array]}
  insureds={[Function Array]}
  policyNumber={[Function String]}
  riskState={[Function String]}
  termEffectiveDate={[Function String]}
  termExpirationDate={[Function String]}
  toggle={[Function EventEmitter_]}
  vehicle={[Function Object]}
>
  <div
    class="poi-card"
  >
    <ion-grid>
      <ion-row
        class="row--heading"
      >
        <ion-col
          class="ion-text-start card-logo"
          size="auto"
        >
          <img
            src="/assets/csaa-mobile/vectors/aaa-logo.svg"
          />
        </ion-col>
        <ion-col
          size="10"
        >
          <p>
            CSAA Fire & Casualty Insurance Company / NAIC #10921
          </p>
          <p>
            PO Box 24511, Oakland, CA 94623-9865
          </p>
          <p>
            800-207-3618
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--title"
      >
        <ion-col>
          <h1>
            OK SECURITY VERIFICATION FORM
          </h1>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              NAMED INSURED
            </span>
            <span
              data-test="named-insured"
            >
               Jean Brandos 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              POLICY #
            </span>
            <span
              data-test="policy-card-number"
            >
               WYSS910014303 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              VEHICLE YEAR / MAKE / MODEL
            </span>
            <span
              data-test="vehicle-title"
            >
               Lamborghini Gallardo 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              EFFECTIVE DATE
            </span>
            <span
              data-test="effective-date"
            >
               May 17, 2021 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              VEHICLE ID#
            </span>
            <span
              data-test="vehicle-Id"
            >
               WAUKJAFM8AA061319 
            </span>
          </p>
        </ion-col>
        <ion-col
          size="4"
        >
          <p>
            <span>
              EXPIRATION DATE
            </span>
            <span
              data-test="expiration-date"
            >
               Nov 17, 2021 
            </span>
          </p>
        </ion-col>
      </ion-row>
      <ion-row
        class="row--content"
      >
        <ion-col
          size="8"
        >
          <p>
            <span>
              EXCLUDED DRIVERS
            </span>
            <span
              data-test="excluded-drivers"
            >
               Judy Quinn 
            </span>
          </p>
        </ion-col>
      </ion-row>
    </ion-grid>
  </div>
</csaa-auto-id-card>
`;
